tests/test_association_db_sqlite.py

```diff
@@ -0,0 +1,266 @@
+#
+# LSST Data Management System
+# Copyright 2017 LSST/AURA.
+#
+# This product includes software developed by the
+# LSST Project (http://www.lsst.org/).
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <http://www.lsstcorp.org/LegalNotices/>.
+#
+
+from __future__ import absolute_import, division, print_function
+
+import numpy as np
+import unittest
+
+from lsst.ap.association import *
+from lsst.afw.coord import Coord
+import lsst.afw.geom as afwGeom
+import lsst.utils.tests
+from test_dia_collection import create_test_dia_objects
+from test_dia_object import create_test_dia_sources
+
+
+class TestAssociationDBSqlite(unittest.TestCase):
+
+    def setup(self):
+        pass
+
+    def tearDown(self):
+        pass
+
+    def test_init(self):
+        """ Test instatiation of the task.
+        """
+        assoc_db = AssociationDBSqliteTask()
+        assoc_db.close()
+
+    def test_create_tables(self):
+        """ Test the creation of an empty table.
+        """
+        assoc_db = AssociationDBSqliteTask()
+        assoc_db.create_tables()
+        assoc_db._commit()
+        assoc_db.close()
+
+    def test_load(self):
+        """ Test loading of DIAObjects
+        """
+        dia_objects = create_test_dia_objects(
+            n_objects=5, n_src=2, increment_degrees=0.04, scatter_arcsec=0.0)
+        dia_collection = DIAObjectCollection(dia_objects)
+        assoc_db = AssociationDBSqliteTask()
+        assoc_db.create_tables()
+
+        assoc_db.store(dia_collection, True)
+
+        ctr_point = Coord(
+            afwGeom.Angle(0.08, units=afwGeom.degrees),
+            afwGeom.Angle(0.08, units=afwGeom.degrees))
+        output_dia_collection = assoc_db.load(
+            ctr_point, afwGeom.Angle(0.2))
+
+        for obj_idx in xrange(5):
```

Comment:
Have you run this on Python 3?

Reply:
Not yet. Sorry about the xrange, Tim. Force of habit.